pkg/resource/defaultclass.go

```diff
@@ -41,44 +42,48 @@ var logDefaultClass = logging.Logger.WithName("controller").WithValues("controll
 
 // Error strings
 const (
-	errFailedList             = "unable to list default resource classes"
-	errNoDefaultClass         = "unable to locate a default resource class for claim kind"
-	errMultipleDefaultClasses = "multiple default classes defined for claim kind"
+	errFailedList             = "unable to list policies for claim kind"
+	errFailedPolicyConversion = "unable to convert located policy to coreect kind"
+	errNoPolicies             = "unable to locate a policy that specifies a default class for claim kind"
+	errMultiplePolicies       = "multiple policies that specify a default class defined for claim kind"
 )
 
+// A PolicyKind contains the type metadata for a kind of policy.
+type PolicyKind schema.GroupVersionKind
+
+// A PolicyListKind contains the type metadata for a kind of policy.
+type PolicyListKind schema.GroupVersionKind
+
 // DefaultClassReconciler reconciles resource claims to the
-// default resource class for their given kind. Predicates
-// ensure that only claims with no resource class reference
-// are reconciled.
+// default resource class for their given kind according to existing
+// policies. Predicates ensure that only claims with no resource class
+// reference are reconciled.
 type DefaultClassReconciler struct {
-	client   client.Client
-	newClaim func() Claim
-	options  *client.ListOptions
+	client        client.Client
+	newClaim      func() Claim
+	newPolicy     func() Policy
+	newPolicyList func() PolicyList
 }
 
 // NewDefaultClassReconciler creates a new DefaultReconciler for the claim kind
-func NewDefaultClassReconciler(m manager.Manager, of ClaimKind) *DefaultClassReconciler {
+func NewDefaultClassReconciler(m manager.Manager, of ClaimKind, by PolicyKind, byList PolicyListKind) *DefaultClassReconciler {
 	nc := func() Claim { return MustCreateObject(schema.GroupVersionKind(of), m.GetScheme()).(Claim) }
+	np := func() Policy { return MustCreateObject(schema.GroupVersionKind(by), m.GetScheme()).(Policy) }
+	npl := func() PolicyList {
+		return MustCreateObject(schema.GroupVersionKind(byList), m.GetScheme()).(PolicyList)
+	}
 
-	// Panic early if we've been asked to reconcile a claim that has
+	// Panic early if we've been asked to reconcile a claim, policy, or polict list that has
 	// not been registered with our controller manager's scheme.
 	_ = nc()
-
-	gk := strings.ToLower(schema.GroupVersionKind(of).GroupKind().String())
-
-	// Create list options query that will be used to search
-	// for resource class that is default for claim kind.
-	options := &client.ListOptions{}
-	if err := options.SetLabelSelector(gk + "/default=" + "true"); err != nil {
-		// Panic if unable to set label selector or else panic will occur
-		// when returned reconciler is invoked.
-		panic(err)
-	}
+	_ = np()
+	_ = npl()
```

Comment:
The other variants of this pattern call these all on the same line, i.e.:

```go
_, _, _ = nc(), np(), npl()
```

Neither way is objectively better, but I'd prefer we did it consistently.